pkg/provider/azure_loadbalancer_healthprobe_test.go

```diff
@@ -197,3 +203,144 @@ func TestFindProbe(t *testing.T) {
 		})
 	}
 }
+
+func TestShouldKeepSharedProbe(t *testing.T) {
```

Comment:
Thanks adding the unit tests. Since this is an important feature, could you also add an e2e test case?

Reply:
We need to create a new prowjob for this feature because we need to modify the azure.json. Is it worth adding an entire pipeline and running all our test cases (most irrelevant to this feature) just to test this new behavior?